src/amo/components/UserProfileEdit/index.js

```diff
@@ -0,0 +1,475 @@
+    event.preventDefault();
+
+    const { dispatch, errorHandler, user } = this.props;
+    const {
+      biography,
+      displayName,
+      homepage,
+      location,
+      occupation,
+      username,
+    } = this.state;
+
+    invariant(user, 'user is required');
+
+    dispatch(editUserAccount({
+      errorHandlerId: errorHandler.id,
+      userFields: {
+        biography,
+        display_name: displayName,
+        homepage,
+        location,
+        occupation,
+        username,
+      },
+      userId: user.id,
+    }));
+  }
+
+  getFormValues(user: UserType | null): FormValues {
+    const defaultFormValues = {
+      biography: '',
+      displayName: '',
+      homepage: '',
+      location: '',
+      occupation: '',
+      username: this.props.username,
+    };
+
+    if (!user) {
+      return defaultFormValues;
+    }
+
+    const {
+      biography,
+      display_name: displayName,
+      homepage,
+      location,
+      occupation,
+      username,
+    } = user;
+
+    return {
+      biography,
+      displayName,
+      homepage,
+      location,
+      occupation,
+      username,
+    };
+  }
+
+  preventSubmit() {
+    const { user, isEditing } = this.props;
+    const { username } = this.state;
+
+    return !user || isEditing || !username || (username &&
+      username.trim() === '');
+  }
+
+  render() {
+    const {
+      currentUser,
+      errorHandler,
+      hasEditPermission,
+      i18n,
+      isEditing,
+      user,
+      username,
+    } = this.props;
+
+    if (!currentUser || (currentUser && user && !hasEditPermission)) {
+      return <NotFound />;
+    }
+
+    const isEditingCurrentUser = currentUser && user ? currentUser.id ===
+      user.id : false;
+
+    return (
+      <div className="UserProfileEdit">
+        {user && (
+          <Helmet>
+            <title>
+              {i18n.sprintf(
+                i18n.gettext('User Profile for %(user)s'),
+                { user: user.displayName }
+              )}
+            </title>
+          </Helmet>
+        )}
+
+        <Card className="UserProfileEdit-user-links">
+          <ul>
+            <li>
+              <Link to={`/user/${username}/`}>
+                {isEditingCurrentUser ?
+                  i18n.gettext('View My Profile') :
+                  i18n.gettext(`View user's profile`)
+                }
+              </Link>
+            </li>
+            <li>
+              {isEditingCurrentUser ?
+                i18n.gettext('Edit My Profile') :
+                i18n.gettext(`Edit user's profile`)
+              }
+            </li>
+          </ul>
+        </Card>
+
+        <form className="UserProfileEdit-form" onSubmit={this.onSubmit}>
+          <div className="UserProfileEdit-form-messages">
+            {errorHandler.renderErrorIfPresent()}
+
+            {this.state.displaySuccessMessage && (
+              <Notice type="success">
+                {i18n.gettext('Profile successfully updated.')}
```

Comment:
I think we tend to leave out the period at the end of confirmation messages. For example, in [AddAddonToCollection](https://github.com/mozilla/addons-frontend/blob/master/src/amo/components/AddAddonToCollection/index.js#L233).